docs/src/types/vectors.md

```diff
@@ -0,0 +1,183 @@
+# Vectors
+
+## Passing in vectors
+You can pass a Rust std::vec::Vec into your contract function transparently. The following code calls a sway function which accepts and then returns a `Vec<SomeStruct<u32>>`.
+
+```rust,ignore
+{{#include ../../../packages/fuels/tests/harness.rs:passing_in_vec}}
+```
+
+You can use a vector just like you would use any other type -- i.e. a `[Vec<u32>; 2]` or a `SomeStruct<Vec<Bits256>>` etc.
+
+## Returning vectors
+There is a mandatory extra step to returning vectors -- you need to `log` every element before returning the vector.
+
+You must not mix any unrelated `log`s once you start logging vector elements. 
+
+These logs must be the last ones you make.
```

Comment:
If we're logging the elements, why are we also returning in it? Is there some magic happening to combine the logs and the return?

Reply:
The only way we know when somebody wants to return something from a function is if they actually return it.

When they do we will generate a function that has that return type -- i.e. a `Vec<...>`

But the return data isn't enough to populate the vector... so it is enriched with the logs.